app/src/androidTest/java/com/android/bookswap/ui/map/MapScreenTest.kt

```diff
@@ -0,0 +1,96 @@
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.assertIsNotDisplayed
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onAllNodesWithTag
+import androidx.compose.ui.test.onNodeWithTag
+import com.android.bookswap.data.BookLanguages
+import com.android.bookswap.data.DataBook
+import java.util.UUID
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class MapScreenTest {
+  val user =
+      listOf(
+          TempUser(
+              latitude = 0.0,
+              longitude = 0.0,
+              listBook =
+                  listOf(
+                      DataBook(
+                          uuid = UUID.randomUUID(),
+                          title = "Book 1",
+                          author = "Author 1",
+                          description = "Description of Book 1",
+                          rating = 5,
+                          photo = "url_to_photo_1",
+                          language = BookLanguages.ENGLISH,
+                          isbn = "123-456-789"),
+                      DataBook(
+                          uuid = UUID.randomUUID(),
+                          title = "Book 2",
+                          author = "Author 2",
+                          description = "Description of Book 2",
+                          rating = 4,
+                          photo = "url_to_photo_2",
+                          language = BookLanguages.FRENCH,
+                          isbn = "234-567-890"),
+                      DataBook(
+                          uuid = UUID.randomUUID(),
+                          title = "Book 3",
+                          author = "Author 3",
+                          description = "Description of Book 3",
+                          rating = 3,
+                          photo = "url_to_photo_3",
+                          language = BookLanguages.GERMAN,
+                          isbn = "345-678-901"))))
+
+  val userWithoutBooks = listOf(TempUser(latitude = 1.0, longitude = 1.0, listBook = emptyList()))
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before fun setUp() {}
+
+  @Test
+  fun displayAllComponents() {
+    composeTestRule.setContent { MapScreen(user, user[0]) }
+    composeTestRule.onNodeWithTag("mapScreen").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("mapGoogleMap").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("mapBoxMarker").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("mapBoxMarkerList").assertIsDisplayed()
+    composeTestRule.onAllNodesWithTag("mapBoxMarkerListBox").assertCountEquals(3)
+    composeTestRule.onAllNodesWithTag("mapBoxMarkerListBoxTitle").assertCountEquals(3)
+    composeTestRule.onAllNodesWithTag("mapBoxMarkerListBoxAuthor").assertCountEquals(3)
+    composeTestRule.onAllNodesWithTag("mapBoxMarkerListDivider").assertCountEquals(2)
+  }
+
+  @Test
+  fun noMarkerDisplayedForUserWithoutBooks() {
```

Comment:
Don't hesitate to use full sentence for test function name : 
```kotlin
@Test
fun `no marker is displayed when there is users without books`()  {
}
```

Reply:
I cannot do that as I get an error if I do that in a file in the androidTest folder. I get this error :
"Identifier not allowed in Android projects"
I checked and it also do that in the bootcamp for the androidTest folder. I don't want to spend time to find why it is like this as this is a very minor problem